### PR TITLE
better error messages for blaze version mismatch

### PR DESCRIPTION
**Pull request by hhuuggoo**

### CODE REVIEW

File: bokeh/server/blaze/config.py [hunk 1]
```diff
@@ -28,18 +32,47 @@
...
+        "error loading hdfstore - perhaps your version of blaze or into is too old"
+    )
+
+
 
 data = dict(uniform=uniform,
             gauss=gauss,
             bivariate=bivariate,
-            aapl=aapl,
             array=arr,
```
Comment (damianavila) [inline bokeh/server/blaze/config.py]: don't you need to remove this line too?

**Comment by damianavila** [comment]
LGTM, waiting for TravisCI...

**Comment by damianavila** [comment]
OK, I can not trigger the warn... these are my logs:
```
(py34devel)[damian@damian-S400CA][bokeh](pr/1905)$ ./bokeh-server --backend=memory

    Bokeh Server Configuration
    ==========================
    python version : 3.4.2
    bokeh version  : 0.8.0-rc-1297302-5-gaffe76b
    listening      : 127.0.0.1:5006
    backend        : memory
    python options : debug:OFF, verbose:OFF, filter-logs:OFF, multi-user:OFF
    js options     : splitjs:OFF, debugjs:OFF
    
/media/datos/Desarrollos/bokeh/bokeh/server/blaze/__init__.py:19: UserWarning: could not import multiuser blaze server No module named 'blaze'.  This is fine if you do not intend to use blaze capabilities in the bokeh server
  warnings.warn(msg)
^CShutting down bokeh-server ...
(py34devel)[damian@damian-S400CA][bokeh](pr/1905)$ conda install -c bokeh multiuserblazeserver
Fetching package metadata: ............
Solving package specifications: ..............
Package plan for installation in environment /home/damian/miniconda3/envs/py34devel:

The following NEW packages will be INSTALLED:

    blaze:                0.7.2-np19py34_0   
    blaze-core:           0.7.2-np19py34_0   
    into:                 0.2.1-np19py34_0   
    multiuserblazeserver: 0.2-py34_0_gfa58b9f

The following packages will be UPDATED:

    bcolz:                0.7.2.dev-np19py34_0 --> 0.8.0-np19py34_0   
    h5py:                 2.3.1-np19py34_1     --> 2.4.0-np19py34_0   
    psutil:               2.1.1-py34_0         --> 2.2.0-py34_0       

Proceed ([y]/n)? y

Unlinking packages ...
[      COMPLETE      ] |##########################################################################################################################################| 100%
Linking packages ...
[      COMPLETE      ] |##########################################################################################################################################| 100%
(py34devel)[damian@damian-S400CA][bokeh](pr/1905)$ ./bokeh-server --backend=memory

    Bokeh Server Configuration
    ==========================
    python version : 3.4.2
    bokeh version  : 0.8.0-rc-1297302-5-gaffe76b
    listening      : 127.0.0.1:5006
    backend        : memory
    python options : debug:OFF, verbose:OFF, filter-logs:OFF, multi-user:OFF
    js options     : splitjs:OFF, debugjs:OFF
    
^CShutting down bokeh-server ...
Closing remaining open files:/media/datos/Desarrollos/bokeh/bokeh/server/tests/data/CensusTracts.hdf5...done/media/datos/Desarrollos/bokeh/bokeh/server/tests/data/AAPL.hdf5...done
(py34devel)[damian@damian-S400CA][bokeh](pr/1905)$ conda list blaze
# packages in environment at /home/damian/miniconda3/envs/py34devel:
#
blaze                     0.7.2                np19py34_0  
blaze-core                0.7.2                np19py34_0  
multiuserblazeserver      0.2             py34_0_gfa58b9f  
(py34devel)[damian@damian-S400CA][bokeh](pr/1905)$ conda remove blaze blaze-core --yes
Fetching package metadata: ..........

Package plan for package removal in environment /home/damian/miniconda3/envs/py34devel:

The following packages will be REMOVED:

    blaze:      0.7.2-np19py34_0
    blaze-core: 0.7.2-np19py34_0

Unlinking packages ...
[      COMPLETE      ] |##########################################################################################################################################| 100%
(py34devel)[damian@damian-S400CA][bokeh](pr/1905)$ conda install blaze=0.7.0
Fetching package metadata: ..........
Solving package specifications: .
Package plan for installation in environment /home/damian/miniconda3/envs/py34devel:

The following NEW packages will be INSTALLED:

    blaze:  0.7.0-np19py34_0    

The following packages will be DOWNGRADED:

    bcolz:  0.8.0-np19py34_0 --> 0.7.2.dev-np19py34_0
    h5py:   2.4.0-np19py34_0 --> 2.3.1-np19py34_1    
    into:   0.2.1-np19py34_0 --> 0.1.3-np19py34_0    
    psutil: 2.2.0-py34_0     --> 2.1.1-py34_0        

Proceed ([y]/n)? y

Unlinking packages ...
[      COMPLETE      ] |##########################################################################################################################################| 100%
Linking packages ...
[      COMPLETE      ] |##########################################################################################################################################| 100%
(py34devel)[damian@damian-S400CA][bokeh](pr/1905)$ conda list blaze
# packages in environment at /home/damian/miniconda3/envs/py34devel:
#
blaze                     0.7.0                np19py34_0  
multiuserblazeserver      0.2             py34_0_gfa58b9f  
(py34devel)[damian@damian-S400CA][bokeh](pr/1905)$ ./bokeh-server --backend=memory

    Bokeh Server Configuration
    ==========================
    python version : 3.4.2
    bokeh version  : 0.8.0-rc-1297302-5-gaffe76b
    listening      : 127.0.0.1:5006
    backend        : memory
    python options : debug:OFF, verbose:OFF, filter-logs:OFF, multi-user:OFF
    js options     : splitjs:OFF, debugjs:OFF
    
Traceback (most recent call last):
  File "/home/damian/miniconda3/envs/py34devel/lib/python3.4/site-packages/h5py/_hl/files.py", line 99, in make_fid
    fid = h5f.open(name, h5f.ACC_RDWR, fapl=fapl)
  File "h5f.pyx", line 71, in h5py.h5f.open (h5py/h5f.c:1797)
OSError: Unable to open file (Unable to open file: name = 'hdfstore:///media/datos/desarrollos/bokeh/bokeh/server/tests/data/aapl.hdf5', errno = 2, error message = 'no such file or directory', flags = 1, o_flags = 2)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/damian/miniconda3/envs/py34devel/lib/python3.4/site-packages/h5py/_hl/files.py", line 102, in make_fid
    fid = h5f.open(name, h5f.ACC_RDONLY, fapl=fapl)
  File "h5f.pyx", line 71, in h5py.h5f.open (h5py/h5f.c:1797)
OSError: Unable to open file (Unable to open file: name = 'hdfstore:///media/datos/desarrollos/bokeh/bokeh/server/tests/data/aapl.hdf5', errno = 2, error message = 'no such file or directory', flags = 0, o_flags = 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/media/datos/Desarrollos/bokeh/bokeh/server/blaze/config.py", line 37, in <module>
    aapl = resource("hdfstore://%s::__data__" % path)
  File "/home/damian/miniconda3/envs/py34devel/lib/python3.4/site-packages/into/regex.py", line 64, in __call__
    return self.dispatch(s)(s, *args, **kwargs)
  File "/home/damian/miniconda3/envs/py34devel/lib/python3.4/site-packages/into/resource.py", line 20, in resource_split
    return resource(uri, other, *args, **kwargs)
  File "/home/damian/miniconda3/envs/py34devel/lib/python3.4/site-packages/into/regex.py", line 64, in __call__
    return self.dispatch(s)(s, *args, **kwargs)
  File "/home/damian/miniconda3/envs/py34devel/lib/python3.4/site-packages/into/backends/h5py.py", line 122, in resource_hdf5
    f = h5py.File(uri)
  File "/home/damian/miniconda3/envs/py34devel/lib/python3.4/site-packages/h5py/_hl/files.py", line 222, in __init__
    fid = make_fid(name, mode, userblock_size, fapl)
  File "/home/damian/miniconda3/envs/py34devel/lib/python3.4/site-packages/h5py/_hl/files.py", line 104, in make_fid
    fid = h5f.create(name, h5f.ACC_EXCL, fapl=fapl, fcpl=fcpl)
  File "h5f.pyx", line 90, in h5py.h5f.create (h5py/h5f.c:1975)
OSError: Unable to create file (Unable to open file: name = 'hdfstore:///media/datos/desarrollos/bokeh/bokeh/server/tests/data/aapl.hdf5', errno = 2, error message = 'no such file or directory', flags = 15, o_flags = c2)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "./bokeh-server", line 7, in <module>
    bokeh.server.run()
  File "/media/datos/Desarrollos/bokeh/bokeh/server/__init__.py", line 177, in run
    start_server(args)
  File "/media/datos/Desarrollos/bokeh/bokeh/server/__init__.py", line 181, in start_server
    start.start_simple_server(args)
  File "/media/datos/Desarrollos/bokeh/bokeh/server/start.py", line 64, in start_simple_server
    register_blueprint()
  File "/media/datos/Desarrollos/bokeh/bokeh/server/configure.py", line 95, in register_blueprint
    blaze_blueprint = get_mbs_blueprint(config_file=server_settings.blaze_config)
  File "/media/datos/Desarrollos/bokeh/bokeh/server/blaze/__init__.py", line 15, in get_blueprint
    mbs.app.setup_app(config_file=config_file)
  File "/home/damian/miniconda3/envs/py34devel/lib/python3.4/site-packages/mbs/app.py", line 12, in setup_app
    settings.from_pyfile(config_file)
  File "/home/damian/miniconda3/envs/py34devel/lib/python3.4/site-packages/mbs/settings.py", line 54, in from_pyfile
    mod = imp.load_source(name, fname)
  File "/home/damian/miniconda3/envs/py34devel/lib/python3.4/imp.py", line 171, in load_source
    module = methods.load()
  File "<frozen importlib._bootstrap>", line 1220, in load
  File "<frozen importlib._bootstrap>", line 1200, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 1129, in _exec
  File "<frozen importlib._bootstrap>", line 1471, in exec_module
  File "<frozen importlib._bootstrap>", line 321, in _call_with_frames_removed
  File "/media/datos/Desarrollos/bokeh/bokeh/server/blaze/config.py", line 40, in <module>
    log.error(e.message)
AttributeError: 'OSError' object has no attribute 'message'
(py34devel)[damian@damian-S400CA][bokeh](pr/1905)$ conda update into
Fetching package metadata: ..........
Solving package specifications: .
Package plan for installation in environment /home/damian/miniconda3/envs/py34devel:

The following packages will be UPDATED:

    into: 0.1.3-np19py34_0 --> 0.2.2-np19py34_0

Proceed ([y]/n)? y  

Unlinking packages ...
[      COMPLETE      ] |##########################################################################################################################################| 100%
Linking packages ...
[      COMPLETE      ] |##########################################################################################################################################| 100%
(py34devel)[damian@damian-S400CA][bokeh](pr/1905)$ ./bokeh-server --backend=memory

    Bokeh Server Configuration
    ==========================
    python version : 3.4.2
    bokeh version  : 0.8.0-rc-1297302-5-gaffe76b
    listening      : 127.0.0.1:5006
    backend        : memory
    python options : debug:OFF, verbose:OFF, filter-logs:OFF, multi-user:OFF
    js options     : splitjs:OFF, debugjs:OFF
    
^CShutting down bokeh-server ...
Closing remaining open files:/media/datos/Desarrollos/bokeh/bokeh/server/tests/data/AAPL.hdf5...done/media/datos/Desarrollos/bokeh/bokeh/server/tests/data/CensusTracts.hdf5...done
(py34devel)[damian@damian-S400CA][bokeh](pr/1905)$ conda update into=0.2.1
Error: Invalid package name: 'into=0.2.1'
(py34devel)[damian@damian-S400CA][bokeh](pr/1905)$ conda install into=0.2.1
Fetching package metadata: ..........
Solving package specifications: .
Package plan for installation in environment /home/damian/miniconda3/envs/py34devel:

The following packages will be DOWNGRADED:

    into: 0.2.2-np19py34_0 --> 0.2.1-np19py34_0

Proceed ([y]/n)? y

Unlinking packages ...
[      COMPLETE      ] |##########################################################################################################################################| 100%
Linking packages ...
[      COMPLETE      ] |##########################################################################################################################################| 100%
(py34devel)[damian@damian-S400CA][bokeh](pr/1905)$ ./bokeh-server --backend=memory

    Bokeh Server Configuration
    ==========================
    python version : 3.4.2
    bokeh version  : 0.8.0-rc-1297302-5-gaffe76b
    listening      : 127.0.0.1:5006
    backend        : memory
    python options : debug:OFF, verbose:OFF, filter-logs:OFF, multi-user:OFF
    js options     : splitjs:OFF, debugjs:OFF
    
^CShutting down bokeh-server ...
Closing remaining open files:/media/datos/Desarrollos/bokeh/bokeh/server/tests/data/AAPL.hdf5...done/media/datos/Desarrollos/bokeh/bokeh/server/tests/data/CensusTracts.hdf5...done
(py34devel)[damian@damian-S400CA][bokeh](pr/1905)$ 
```

The closer I can get to the error is installing multiuserblaze server, and then downgrade to blaze 0.7.0 (with into 0.1.3), but even in this case the warn seems to not work, besides the fact, it seems the same error...

**Comment by hhuuggoo** [comment]
Hey @damianavila thanks!  This line explains it

```
  File "/media/datos/Desarrollos/bokeh/bokeh/server/blaze/config.py", line 40, in <module>
    log.error(e.message)
AttributeError: 'OSError' object has no attribute 'message'
```

I just wanted to log one line of the exception, rather than the entire stack trace - what's the better way to do this?

**Comment by damianavila** [comment]
ok, with this diff...

```shell
(py34devel)[damian@damian-S400CA][bokeh](pr/1905)$ git diff
diff --git a/bokeh/server/blaze/config.py b/bokeh/server/blaze/config.py
index 5336ca4..0c9f33c 100644
--- a/bokeh/server/blaze/config.py
+++ b/bokeh/server/blaze/config.py
@@ -37,7 +37,7 @@ try:
     aapl = resource("hdfstore://%s::__data__" % path)
 except Exception as e:
     aapl = None
-    log.error(e.message)
+    log.error(e)
     warnings.warn(
         "error loading hdfstore - perhaps your version of blaze or into is too old"
     )
@@ -47,7 +47,7 @@ try:
     arr = resource(path + "::" + "array")
 except Exception as e:
     arr = None
-    log.error(e.message)
+    log.error(e)
     warnings.warn(
         "error loading hdf5 - perhaps your version of blaze or into is too old"
     )
@@ -58,7 +58,7 @@ try:
     census = resource("hdfstore://%s::__data__" % path)
 except Exception as e:
     census = None
-    log.error(e.message)
+    log.error(e)
     warnings.warn(
         "error loading hdfstore - perhaps your version of blaze or into is too old"
     )
```

I get the proper response:

```shell
(py34devel)[damian@damian-S400CA][bokeh](pr/1905)$ ./bokeh-server --backend=memory

    Bokeh Server Configuration
    ==========================
    python version : 3.4.2
    bokeh version  : 0.8.0-rc-1297302-5-gaffe76b-dirty
    listening      : 127.0.0.1:5006
    backend        : memory
    python options : debug:OFF, verbose:OFF, filter-logs:OFF, multi-user:OFF
    js options     : splitjs:OFF, debugjs:OFF
    
ERROR:_mbs_configuration:Unable to create file (Unable to open file: name = 'hdfstore:///media/datos/desarrollos/bokeh/bokeh/server/tests/data/aapl.hdf5', errno = 2, error message = 'no such file or directory', flags = 15, o_flags = c2)
/media/datos/Desarrollos/bokeh/bokeh/server/blaze/config.py:42: UserWarning: error loading hdfstore - perhaps your version of blaze or into is too old
  "error loading hdfstore - perhaps your version of blaze or into is too old"
ERROR:_mbs_configuration:Cannot create dshape from object of type <class 'NoneType'>
/media/datos/Desarrollos/bokeh/bokeh/server/blaze/config.py:52: UserWarning: error loading hdf5 - perhaps your version of blaze or into is too old
  "error loading hdf5 - perhaps your version of blaze or into is too old"
ERROR:_mbs_configuration:Unable to create file (Unable to open file: name = 'hdfstore:///media/datos/desarrollos/bokeh/bokeh/server/tests/data/censustracts.hdf5', errno = 2, error message = 'no such file or directory', flags = 15, o_flags = c2)
/media/datos/Desarrollos/bokeh/bokeh/server/blaze/config.py:63: UserWarning: error loading hdfstore - perhaps your version of blaze or into is too old
  "error loading hdfstore - perhaps your version of blaze or into is too old"
```

Is enough info for you?

**Comment by hhuuggoo** [comment]
yes - let's add your diff onto this branch and then merge

File: bokeh/server/blaze/config.py [hunk 2]
```diff
@@ -28,18 +32,46 @@
...
 path = join(dirname(bokeh.server.tests.__file__), 'data', 'AAPL.hdf5')
-aapl = resource("hdfstore://%s::__data__" % path)
+
+try:
+    aapl = resource("hdfstore://%s::__data__" % path)
+except Exception as e:
+    aapl = None
+    log.error(e)
+    warnings.warn(
+        "error loading hdfstore - perhaps your version of blaze or into is too old"
```
Comment (bryevdv) [inline bokeh/server/blaze/config.py]: prefer stronger wording: "your version of Blaze is too old, or incompatible"

Comment (damianavila) [inline bokeh/server/blaze/config.py]: ok, I will change it...

Comment (damianavila) [inline bokeh/server/blaze/config.py]: Changed, I will merge it in a few hours if there is not more comments.

**Comment by damianavila** [comment]
OK, merging this one in a few hour unless I get another request...

**Pull request merged**